### PR TITLE
Improve code

### PR DESCRIPTION
**Pull request by vkhramtsov**
- [x] Tests improvements (use proper methods from php unit instead of php constructions and checks for true and false)
- [x] Performance improved with using `static` for closures
- [x] Optimize regular expressions (replace `\w\d` with just `\w` (because `\w` already include `\d)`, remove excessive escaping, use character classes (`[vl]`) instead of or (`v|l`), remove `i` flag for patterns without alphabetical symbols)
- [x] Remove obsolete imports
- [x] Remove obsolete test
- [x] Fix phpdoc in tests

### CODE REVIEW

**Comment by vkhramtsov** [comment]
I see some incomplete tests and one error 504. Should I investigate them? Locally I have only one incomplete test

File: src/Autoloader.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?php
-spl_autoload_register(function ($class) {
-    if (!preg_match('~Embera~i', $class)) {
-        return ;
+spl_autoload_register(static function ($class) {
+    if (false === stripos($class, 'Embera')) {
```
Comment (QWp6t) [inline src/Autoloader.php]: but why yoda condition? lol

Comment (vkhramtsov) [reply src/Autoloader.php]: And why not? I can easily change this in case its required

**Comment by mpratt** [comment]
Thanks @vkhramtsov for the PR!
I'll be taking a look at it tonight! Dont worry about the failing tests, I'll take care of it, travis sometimes has issues running the code, specially making requests to some providers, so dont worry. Sometimes the tests run fine locally, but travis complains about something.

**Comment by vkhramtsov** [comment]
@mpratt looking forward for your feedback

**Comment by mpratt** [comment]
Thanks @vkhramtsov 
Nice work, great catches! Merged and soon I'll be tagging 2.0.2

**Comment by vkhramtsov** [comment]
Perfect. I'm waiting for it because I want to upgrade your package to version 2 in our project

**Comment by mpratt** [comment]
Yeah, don't worry. I just added support for another provider (Avocode), Im just waiting for travis to run the tests and see if I need to update something else before tagging.

**Pull request merged**